opentracing/src/main/java/opentracing/Tracer.java

```diff
@@ -31,7 +36,9 @@
    *                   .setTag("lucky_number", 42);
    * }</pre>
    */
-  Span startTrace(String operationName);
+  Span startSpan(String operationName, Optional<Span> parent);
+
+  <T> Injector<T> getInjector(Class<T> carrierType);
```

Comment:
@michaelsembwever @yurishkuro 

I was having trouble sleeping last night and found myself thinking about this API as well... I had a partially-overlapping idea: namely that there's nothing gained from an abstraction standpoint by having both `getInjector(x)` and `Injector.injectSpan(span, x)`. An `Injector` impl written for `Tracer` implementation `TracerA` cannot be reused for `Tracer` implementation `TracerB`... maybe there are helpers that could be factored out for the benefit of `TracerA` and `TracerB`, but we can't expect the various `Injector` and `Extractor` subclasses to work for an arbitrary `Tracer` implementation.

If we went with the approach here – i.e., no "format" arguments, which does work naturally in Java per previous discussions – the API would look more like

```
public interface Tracer {
  ...

  <T> void injectSpan(Span span, T carrier);
}
```

... or, if we wanted to make things more loosely coupled:

```
public interface Injector {
  <T> void injectSpan(Span span, T carrier);
}

public interface Tracer extends Injector {
  ...
}
```

Thoughts? (cc @tschottdorf since similar things could be done in Go with or without the `format` strings... it seems like we often have these format-specific carrier types in Go and the `reflect` package could be used to reduce the number of concepts in that case)